src/class/InsCertificateValidator/InsCertificateValidator.class.ts

```diff
@@ -0,0 +1,85 @@
+import { util, asn1, pkcs12, pki } from 'node-forge';
+import { IPKCS12Certificate, IINSValidationResponse, INSCertificateValidity, INSValidityAssertion, AssertionType, AssertionStatus } from './InsCertificateValidator.model';
+
+export class InsCertificateValidator {
+  private static _decryptCertificate(pfx: Buffer, passPhrase: string): IPKCS12Certificate {
+    const pfxB64 = pfx.toString('base64');
+
+    const p12Der = util.decode64(pfxB64);
+    const p12Asn1 = asn1.fromDer(p12Der);
+    const p12 = pkcs12.pkcs12FromAsn1(p12Asn1, passPhrase); // raise an error if certif/passphrase combination is invalid
+    const bag =  p12.getBags({
+      bagType: pki.oids.certBag,
+    })[pki.oids.certBag]?.[0]?.cert;
+
+    if (!bag) {
+      throw new Error('The certificate\'s bag is undefined !')
+    }
+
+    const PKCS12Certificate: IPKCS12Certificate = {
+      pfx: pfxB64,
+      subjectCN: bag.subject.getField('CN').value,
+      issuerCN: bag.issuer.getField('CN').value,
+      validity: bag.validity,
+    };
+
+    return PKCS12Certificate;
+  }
+
+  public static validateINSCertificate(pfx: Buffer, passPhrase: string): IINSValidationResponse {
```

Comment:
Le nom de la class gère la déjà le côté INS
```suggestion
  public static validatePKCS12(pfx: Buffer, passPhrase: string): IINSValidationResponse {
```